markdown/ch7.markdown

```diff
@@ -342,12 +342,13 @@ def align(reference, target, cost=cost_mse):
         res = optimize.minimize(cost, p, args=(ref, tgt), method='Powell')
         p = res.x
 
-        print('Pyramid level %i' % n)
-        print('Angle:', np.rad2deg(res.x[0]))
-        print('Offset:', res.x[1:] * 2 ** n)
-        print('Cost function:', res.fun)
-        print('')
+        # print current level, overwriting each time (like a progress bar)
+        print('Level: %i, Angle: %.2f, Offset: (%.1f, %.1f), Cost: %.2f' %
```

Comment:
My immediate thought was: formatting!  But looking at it now, it actually looks worse :)

```
state = {'level': n,
         'angle': np.rad2deg(res.x[0]),
         'offset': (res.x[1] ** 2**n, res.x[2] * 2**n),
         'cost': res.fun}

print('Level: {level:i}, Angle: {angle:.2f}, '
      'Offset: ({offset[0]:.1f}, {offset[1]:.1f}), '
      'Cost: {cost:%.2f'.format(**state))

# or

print('Level: {0:i}, Angle: {angle:.2f}, '
      'Offset: ({2:.1f}, {3:.1f}), Cost: {4:.2f}'.format(
          n, np.rad2deg(res.x[0]),
          res.x[1] ** 2**n, res.x[2] ** 2**n,
          res.fun))
```